.gitignore

```diff
@@ -1,11 +1,15 @@
-node_modules
+# Directories that may be generated by this project
 build
 coverage
-gutenberg.pot
-.vscode
-*.log
-yarn.lock
-gutenberg.zip
+node_modules
 storybook-static
+
+# Files that may be generated by this project
+gutenberg.zip
+
+# Files that may appear in your environment
 .DS_Store
+.vscode
```

Comment:
I see `.vscode` was already here and this is just moving its location, though should it really be here in the first place?

Via https://github.com/WordPress/gutenberg/pull/1382#discussion_r125820857 from a couple of days ago:
> IDE specific entries in `.gitignore` should not be in the repo, users should manage these locally in `~/.gitignore_global`
>
> See also https://core.trac.wordpress.org/ticket/34345